implementations/rust/router/src/router.rs

```diff
@@ -0,0 +1,168 @@
+#![allow(unused)]
+pub mod router {
+    use ockam_message::message::*;
+    use std::collections::HashMap;
+    use std::io::{Error, ErrorKind, Read, Write};
+    use std::net::UdpSocket;
+    use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+    use std::ops::Add;
+    use std::sync::{Arc, Mutex};
+    use std::thread;
```

Comment:
I think we want the router to be `no_std` and have no knowledge of / dependence on transport protocols /  wire encoding / threading / IO etc. 
https://rust-embedded.github.io/book/intro/no-std.html

Reply:
The router has no knowledge of transport. The only reason transport-related stuff is in there is for the tests. The actual router code only knows about addresses and dispatch functions.

What do you mean by "no_std"? If I can't use std, it adds weeks of work to implement OS-independent equivalents for hashmap, threads, and all the transport stuff. What do you propose instead? @mrinalwadhwa